commands.go

```diff
@@ -170,38 +169,86 @@ func (b *bot) mute(m dggchat.Message, s *dggchat.Session) {
 	s.SendMute(parts[1], -1)
 }
 
+// !deletecommand command
+func (b *bot) deleteCommand(m dggchat.Message, s *dggchat.Session) {
+	if !isMod(m.Sender) || !strings.HasPrefix(m.Message, "!deletecommand") {
+		return
+	}
+
+	parts := strings.SplitN(m.Message, " ", 3)
+	if len(parts) < 2 {
+		return
+	}
+
+	cmnd := strings.TrimSpace(parts[1])
+
+	b.commandMutex.Lock()
+	defer b.commandMutex.Unlock()
+
+	for i, c := range b.commands {
+		if strings.EqualFold(c.Call, cmnd) {
+			err := b.DeleteStaticCommands(c)
+			if err != nil {
+				log.Printf("[##] error deleting command: %v", err)
```

Comment:
`log.Printf`(almost) always needs a `\n` at the end. Some more occurences in this code.

Reply:
i never needed /n in printf and sometimes my editor tells me to remove it

Reply:
Figured it out: https://godoc.org/log
"Every log message is output on a separate line: if the message being printed does not end in a newline, the logger will add one."
I'd guess editor telling you to remove something happens in Sprintf commands where the strings might be re-used in another string :thinking: